api/v1alpha3/webhooks.go

```diff
@@ -33,3 +34,20 @@ func aggregateObjErrors(gk schema.GroupKind, name string, allErrs field.ErrorLis
 		allErrs,
 	)
 }
+
+func isValidSSHKey(sshKey *string) field.ErrorList {
+	var allErrs field.ErrorList
+	if sshKey != nil {
+		reg, err := regexp.Compile("[^-A-Za-z0-9-]+")
```

Comment:
Would this have worked as an OpenAPI validation instead, i.e. by setting the kubebuilder annotation on the type?

Reply:
yes i will made those changes

Reply:
I've created https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/1999, maybe we should look at next week.